aten/src/ATen/native/cuda/PowKernel.cu

```diff
@@ -101,30 +101,30 @@ static inline __host__ __device__ B complex_pow_(B base, E exp) {
 }
 #endif
 
-void pow_tensor_tensor_kernel(TensorIterator& iter) {
+void pow_tensor_tensor_kernel(TensorIteratorBase& iter) {
   if (isComplexType(iter.dtype())) {
     AT_DISPATCH_COMPLEX_TYPES(iter.dtype(), "pow_cuda", [&]() {
```

Comment:
The reason for wrong behavior on cuda is probably because dispatch here is on `iter.dtype()` (which is output dtype) and not on `iter.common_dtype()`. cc @mruberry

Reply:
Sounds like a footgun if I've ever heard one

Reply:
It does! Historically `out` participated in type promotion, so this was (almost always) fine, but then it stopped, and suddenly it wasn't.